src/QirRuntime/README.md

```diff
@@ -0,0 +1,115 @@
+2. mkdir build
+3. cd build
+4. cmake -G Ninja -DCMAKE_BUILD_TYPE=Release ..
+5. cmake --build .
+
+Or you can run `build.py` script from the root of the project. The default options for the script are `make debug`.
+
+- (Windows) `python build.py [make/nomake] [debug|release] [ir]`
+- (Linux) `python3 build.py [make/nomake] [debug|release] [ir]`
+
+The scrip will place the build artefacts into `build/[Windows|Linux]/[Debug|Release]` folder.
+
+### Windows pre-reqs
+
+1. Install Clang, Ninja and CMake from the public distros.
+2. Add all three to your/system `%PATH%`.
+3. Install VS 2019 (Clang uses MSVC's standard library on Windows).
+4. [_optional_]
+    1. Either clone LLVM's repo and build LLVM locally to produce your own `opt.exe` and other IR tools or download LLVM's dev package.
+    2. Either modify ProduceIR.cmake to point to these tools, or copy them into a separate folder and add that folder to `%PATH%.` Don't add the whole target build folder to `%PATH%` because it would likely confuse Clang.
+    3. To produce IR targets automatically as part of the build, either pass `ir` option to `build.py` or uncomment setting `${GENERATE_IR}` variable in the root `CMakeLists.txt`.
+
+*Building from Visual Studio and VS Code is **not** supported.
+Running cmake from the editors will likely default to MSVC or clang-cl and fail.*
+
+### Linux via WSL pre-reqs
+
+1. On the host Windows machine install WLS.
+2. In WSL install clang/llvm and the other tools:
+    1. NB: the default version of clang/llvm on WSL's Ubuntu are likely to be too old, it's recommended you get the latest packages.
+    2. $ sudo apt install clang
+    3. $ sudo apt install llvm
+    4. $ sudo apt-get install ninja-build
+    5. Set Clang as the preferred C/C++ compiler:
+        - $ export CC=/usr/bin/clang
+        - $ export CXX=/usr/bin/clang++
+
+See [https://code.visualstudio.com/docs/remote/wsl] on how to use VS Code with WSL.
+
+## Test
+
+To execute all tests, including QIR and managed interop (the interop is Windows only) run `test.py` from the project's root folder:
+
+- (Windows) `python test.py [nobuild] [debug/release]`
+- (Linux) `python3 test.py [nobuild] [debug/release]`
+
+The script will trigger incremental build unless `nobuild` options is specified.
+
+*NB*: the test script currently doesn't detect build failures, so in case of build failure it will run tests against the previously built binaries.
+
+All test binaries and their dependencies are copied by the build into *install* folder: `build/[Windows|Linux]/[Debug|Release]/bin` and should be run from there (otherwise the tests might fail to load the shared libraries they depend on). On **Linux** `test.py` adds the install folder to LD_LIBRARY_PATH for the duration of the script. If you'd like to run the tests directly, add the path for the session manually. On WLS it might look like this: `$export LD_LIBRARY_PATH=/mnt/d/repos/llvm-project/qtracer/build/Linux/Debug/bin:${LD_LIBRARY_PATH}`.
```

Comment:
```suggestion
All test binaries and their dependencies are copied by the build into *install* folder: `build/[Windows|Linux]/[Debug|Release]/bin` and should be run from there (otherwise the tests might fail to load the shared libraries they depend on). On **Linux** `test.py` adds the install folder to LD_LIBRARY_PATH for the duration of the script. If you'd like to run the tests directly, add the path for the session manually. On WSL it might look like this: `$export LD_LIBRARY_PATH=/mnt/d/repos/llvm-project/qtracer/build/Linux/Debug/bin:${LD_LIBRARY_PATH}`.
```